eruditus/lib/discord_util.py

```diff
@@ -0,0 +1,246 @@
+import re
+from datetime import datetime
+from typing import Optional
+
+import aiohttp
+import discord
+
+from config import (
+    CHALLENGE_COLLECTION,
+    CTF_COLLECTION,
+    DBNAME,
+    MAX_CONTENT_SIZE,
+    MONGO,
+    TEAM_NAME,
+)
+from lib.platforms import PlatformCTX, match_platform
+from lib.util import plot_scoreboard
+
+
+async def get_ctf_info(
+    interaction: discord.Interaction, name: Optional[str] = None
+) -> Optional[dict]:
+    # Trying to find ctf by channel category
+    if name is None:
+        ctf = MONGO[DBNAME][CTF_COLLECTION].find_one(
+            {"guild_category": interaction.channel.category_id}
+        )
+
+        if ctf is None:
+            await interaction.followup.send(
+                (
+                    "Run this command from within a CTF channel, or provide the "
+                    "name of the CTF you wish to delete."
+                )
+            )
+            return None
+
+        return ctf
+
+    # Trying to find ctf by provided name
+    ctf = MONGO[DBNAME][f"{CTF_COLLECTION}"].find_one(
+        {"name": re.compile(f"^{re.escape(name.strip())}$", re.IGNORECASE)}
+    )
+    if ctf is None:
+        await interaction.followup.send("No such CTF.")
+        return None
+
+    return ctf
```

Comment:
Since this function is not only used for deleting a CTF, it shouldn't be sending any messages, its task should be limited to returning the CTF dict only.